src/components/Purchase/CheckoutComplete.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useEffect } from 'react';
+import { Container, Paper, Typography, Grid, Box } from '@mui/material';
+import style from './CheckoutComplete.module.css';
+import { useSearchParams } from 'react-router-dom';
+import Loader from '../layout/Loader/Loader';
+import axiosInstance from '../../apis/axiosClient';
+import CheckoutMessageCard from './CheckoutMessageCard';
+
+const ordersEndpoint = '/orders';
+
+const CheckoutComplete = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const clientSecret = searchParams.get('payment_intent_client_secret');
+  const orderId = searchParams.get('orderId');
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [orderData, setOrderData] = useState('');
+  const [isPaymentSuccessful, setIsPaymentSuccessful] = useState(false);
+
+  // Callback function to update isPaymentSuccessful
+  const updatePaymentStatus = (isSuccess) => {
+    setIsPaymentSuccessful(isSuccess);
+  };
+
+  useEffect(() => {
+    const orderDetails = async (id) => {
+      setIsLoading(true);
+      setErrorMessage('');
+      try {
+        const response = await axiosInstance.get(`${ordersEndpoint}/${id}`);
+        setIsLoading(false);
+        setOrderData(response.data?.order);
+        console.log(response.data?.order);
+      } catch (error) {
+        setIsLoading(false);
+        setErrorMessage(
+          `An error occurred during loading order details with id ${id}. ${error.message}. ${error.request.responseText}.`
+        );
+      }
+    };
+    orderDetails(orderId);
+  }, [clientSecret]);
+
+  // Calculate the total number of items purchased
+  let totalItemsPurchased = 0;
+  if (orderData && orderData.orderItems) {
+    orderData.orderItems.forEach((item) => {
+      totalItemsPurchased += item.quantity;
+    });
+  }
+
+  return (
+    <>
+      <Container className={style.mainContainer}>
+        <Grid container spacing={3}>
+          <Grid item xs={12} sm={6}>
+            <CheckoutMessageCard
+              isLoading={isLoading}
+              clientSecret={clientSecret}
+              totalItemsPurchased={totalItemsPurchased}
+              updateStatus={updatePaymentStatus}
+            />
+          </Grid>
+
+          <Grid item xs={12} sm={6}>
+            <Paper className={`${style.paper} ${style.rightPaper}`}>
+              {!isPaymentSuccessful && (
+                <Typography variant="h6">
+                  Payment was unsuccessful! Sorry for the inconvenience.
+                </Typography>
```

Comment:
 f

Reply:
Yes. Good point! The glitch was indeed present.

Reply:
Hi Akos, I think I found another alternative way to fix the loading glitch. I decided to take the initiative and offer my own solution. It seems to me that it has a little less lines of code, however, I can implement your approach if my solution does not look optimal enough. In my approach, instead of creating another loading state, I simply pass the value from isLoading through a prop to the child component. Yes, this is prop drilling, but on the other hand, it seems to me that in this way we add less lines of code. Your opinion on this matter is very interesting to me.

Reply:
I reverted my changes and implemented the previously recommended approach.